packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -86,6 +87,23 @@ contract ERC20Vault is BaseVault {
     error VAULT_INVALID_SRC_CHAIN_ID();
     error VAULT_MESSAGE_NOT_FAILED();
     error VAULT_MESSAGE_RELEASED_ALREADY();
+    error VAULT_ERROR_NATIVE_BURN();
+    error VAULT_ERROR_NATIVE_MINT();
+
+    /// @dev If a native token issuer (e.g.: Circle/Lido) revokes minter role from our contracts
+    /// (once they get the ownership), we won't be able to mint / burn the tokens. But we still want
+    /// to mint these tokens like "USDC ⭀31337" style, but then we need to reset the old mapping.
```

Comment:
Okay, so i'll try to explain the necessity of it via an example!
1. We predeploy the USDC contracts to L2. We call the registry's `changeCustomToken()` to create a mapping between USDC (on L1) and USDC (on. L2).
2. Eventually, when bridge is "safe", Circle needs to have the ownership over the L2 USDC contract(s).
3. For whatever reason (hack in bridge, no trust in Taiko, rival bridges, etc.) Circle will revoke the minter role from our ERC20Vault.

After this, Bob wants to bridge USDC from L1 to L2 via our bridge contract, but it will revert since we do not have minter role anymore. But he also not able to bridge the token into a newly created (BridgedERC20 token named for example `"USDC ⭀31337"` -> This is just a nameing convention we used with BridegdERC20 tokens, this is why i said this in the above example) because such token cannot be created hence the `canonicalToBridged[chainId][canoncalAddr]` will contain the old address.. So we need a mechanism to wipe this data away -> to support further the bridging of the USDC (even tho not 'natively' but via `BridgedERC20`)

Let me know if it makes sense, or not.

Reply:
Instead of saying "But we still want to mint these tokens like "USDC ⭀31337" style", wouldn't it more clear to just say that when that happens it still needs to be possible to bridge tokens using the standard `BridgedERC20`?

Reply:
Yes that would be the ideal i was thinking about it too! :) 
But it requires all the time to “query” if we can bridge native way or not, which is expensive i think. (Bc of that canonicalToBridge mapping for that canonical “occupied” with an address which is not mintable).
Do you have any ideas with which we could do it a cheap and efficient way ? 🤔